cli/cmd/uninstall/uninstall.go

```diff
@@ -345,13 +367,60 @@ func (c *Command) uninstallHelmRelease(releaseName, namespace, releaseType strin
 		}
 	}
 
+	// Delete any custom resources managed by Consul. If they cannot be deleted,
+	// patch the finalizers to be empty on each one.
+	if releaseType == common.ReleaseTypeConsul {
+		uiLogger("Deleting custom resources managed by Consul")
+		crds, err := c.fetchCustomResourceDefinitions()
+		if err != nil {
+			return fmt.Errorf("unable to fetch Custom Resource Definitions for Consul deployment: %v", err)
+		}
+		kindToResource := mapCRKindToResourceName(crds)
+		err = backoff.Retry(func() error {
+			crs, err := c.fetchCustomResources(crds)
+			if err != nil {
+				return err
+			}
+			if len(crs) == 0 {
+				return nil
+			}
+
+			if err = c.deleteCustomResources(crs, kindToResource); err != nil {
+				return err
+			}
+
+			crs, err = c.fetchCustomResources(crds)
+			if err != nil {
+				return err
+			}
+			if len(crs) != 0 {
+				return common.NewDeletionError(fmt.Sprintf("%d custom resources remain after deletion request. Retrying deletion", len(crs)))
+			}
+
+			return nil
+		}, backoff.WithMaxRetries(backoff.NewConstantBackOff(time.Second), 5))
+		if common.IsDeletionError(err) {
+			uiLogger("Patching finalizers on custom resources managed by Consul")
+			crs, err := c.fetchCustomResources(crds)
+			if err != nil {
+				return err
+			}
+
+			if err = c.patchCustomResources(crs, kindToResource); err != nil {
+				return err
+			}
```

Comment:
After we patch CRs, we should have another retry block to check that they are deleted properly for completeness

Reply:
Added! It will print out an error if there is a situation where the CRs cannot be deleted.